### PR TITLE
Add more unit test to write_prepared txns

### PR DESCRIPTION
**Pull request by maysamyabandeh**

### CODE REVIEW

File: utilities/transactions/pessimistic_transaction_db.h [hunk 1]
```diff
@@ -223,13 +229,16 @@ class WritePreparedTxnDB : public PessimisticTransactionDB {
         heap_.pop();
         erased_heap_.pop();
       }
+      while (heap_.empty() && !erased_heap_.empty()) {
+        erased_heap_.pop();
```
Comment (yiwu-arbug) [inline utilities/transactions/pessimistic_transaction_db.h]: TIL std::priority_queue::clear() doesn't exists.

File: utilities/transactions/transaction_test.h [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      delete handles[0];
+    }
+    return s;
+  }
+};
+
+class MySQLStyleTransactionTest : public TransactionTest {};
+
+// TODO(myabandeh): Instantiate the tests with other write policies
+INSTANTIATE_TEST_CASE_P(DBAsBaseDB, TransactionTest,
```
Comment (yiwu-arbug) [inline utilities/transactions/transaction_test.h]: These INSTANTIATE_TEST_CASE_P should be in .cc?

Comment (maysamyabandeh) [reply utilities/transactions/transaction_test.h]: Sure.

File: utilities/transactions/pessimistic_transaction_db.h (hunk 1 above)
Comment (maysamyabandeh) [reply utilities/transactions/pessimistic_transaction_db.h]: Yeah it does not. Not sure what you wanted to imply though.

Comment (yiwu-arbug) [reply utilities/transactions/pessimistic_transaction_db.h]: Just feeling surprised about the std interface.

**Comment by yiwu-arbug** [comment]
Travis got timeout though.

**Comment by ajkr** [comment]
Travis is timing out, can you take a look? E.g., https://travis-ci.org/facebook/rocksdb/jobs/270643329

**Comment by maysamyabandeh** [comment]
@ajkr you can safely ignore this timeout. I will figure a way to signal travis that the test is actually running. It would not be until next week though.

**Pull request closed without merging**